libmy/my_num_to_str.h

```diff
@@ -0,0 +1,31 @@
+#ifndef MY_NUM_TO_STR_H
+#define MY_NUM_TO_STR_H
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <stddef.h>
+
+/*
+ * Format bytes as a null-terminated hex string, with each byte represented as two characters.
+ * Returns a pointer to the newly formatted string.
+ */
+const char *my_bytes_to_hex_str(const uint8_t *src, size_t len, bool is_upper, char *dst);
+
+/*
+ * Format num as hex string, but without a nul terminating byte.
```

Comment:
Since `my_uint16_to_hex_str` calls the NUL-terminating `my_bytes_to_hex_str`, it too NUL-terminates.